app/views/RoomView/index.tsx

```diff
@@ -439,13 +488,60 @@ class RoomView extends React.Component<IRoomViewProps, IRoomViewState> {
 		console.countReset(`${this.constructor.name}.render calls`);
 	}
 
+	canForwardGuest = async () => {
+		const { transferLivechatGuestPermission } = this.props;
+		const permissions = await hasPermission([transferLivechatGuestPermission], this.rid);
+		return permissions[0] as boolean;
+	};
+
+	canPlaceLivechatOnHold = () => {
+		const { livechatAllowManualOnHold } = this.props;
+		const { room } = this.state;
+		return !!(livechatAllowManualOnHold && !room?.lastMessage?.token && room?.lastMessage?.u && !room.onHold);
+	};
+
+	canViewCannedResponse = async () => {
+		const { viewCannedResponsesPermission } = this.props;
+		const permissions = await hasPermission([viewCannedResponsesPermission], this.rid);
+		return permissions[0] as boolean;
+	};
+
+	canReturnQueue = async () => {
+		try {
+			const { returnQueue } = await getRoutingConfig();
+			return returnQueue;
+		} catch {
+			return false;
+		}
+	};
+
+	observeOmnichannel = () => {
+		try {
+			const db = database.active;
+			const subCollection = db
+				.get('subscriptions')
+				.query(Q.where('rid', this.rid as string))
+				.observe();
+			this.subOmnichannel = subCollection.subscribe(data => {
+				if (data[0]) {
+					if (this.subOmnichannel && this.subOmnichannel.unsubscribe) {
+						this.observeRoom(data[0]);
+						this.subOmnichannel.unsubscribe();
+					}
+				}
+			});
+		} catch (e) {
+			console.log("Can't find subscription to observe");
+		}
+	};
```

Comment:
This seems to fix the problem when taking the chat and permissions not being updated on chat messages being sent, not sure of any other way to do it